open-metadata-implementation/access-services/subject-area/subject-area-api/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/properties/objects/graph/Line.java

```diff
@@ -75,16 +79,48 @@ public Line(Line template) {
         this.setGuid(template.getGuid());
         this.setLineType(template.getLineType());
         this.setAdditionalProperties(template.getAdditionalProperties());
+        this.setLineEnds();
     }
 
     public Line(String name) {
-        this.name = name;
-        this.lineType = LineType.Unknown;
+        this(name,LineType.Unknown);
     }
 
     public Line(String name, LineType lineType) {
         this.name = name;
         this.lineType = lineType;
+        this.setLineEnds();
+    }
+    protected void setLineEnds() {
+        OpenMetadataTypesArchiveAccessor archiveAccessor = OpenMetadataTypesArchiveAccessor.getInstance();
+
+        RelationshipDef relationshipDef =archiveAccessor.getRelationshipDefByGuid(this.typeDefGuid);
+        RelationshipEndDef endDef1 = relationshipDef.getEndDef1();
+        RelationshipEndDef endDef2 = relationshipDef.getEndDef2();
+
+        String end1TypeName = endDef1.getEntityType().getName();
+        String end2TypeName = endDef2.getEntityType().getName();
+        // Maybe consider subtypes here? We cannot use the Repositoryhelper isTypeOf because Line is used client side
+        // which does not have access to the repository helper.
+        if (end1TypeName.equals("GlossaryTerm")) {
+            end1TypeName = "Term";
+        }
+        if (end2TypeName.equals("GlossaryTerm")) {
+            end2TypeName = "Term";
```

Comment:
I would prefer use "GlossaryTerm" and remove these conditionals.

Reply:
The Subject Area API does not expose a concept called GlossaryTerm, only Term. I know it is easier internally as it matches the types.

Reply:
I think if we leave GlossaryTerm here, it won't hurt anyone) or it is worth considering revising domain names to map to internal types, as is done with Line
And that help to us in future.

Reply:
this code no longer exists